src/components/Attachment/Attachment.tsx

```diff
@@ -0,0 +1,445 @@
+  AttachmentActions as DefaultAttachmentActions,
+} from './AttachmentActions';
+import { AudioProps, Audio as DefaultAudio } from './Audio';
+import { CardProps, Card as DefaultCard } from './Card';
+import {
+  FileAttachment as DefaultFile,
+  FileAttachmentProps,
+} from './FileAttachment';
+
+import {
+  Gallery as DefaultGallery,
+  ImageComponent as DefaultImage,
+  GalleryProps,
+  ImageProps,
+} from '../Gallery';
+
+import type { Attachment as StreamAttachment } from 'stream-chat';
+
+import type { UnknownType } from '../../../types/types';
+
+export const SUPPORTED_VIDEO_FORMATS = [
+  'video/mp4',
+  'video/ogg',
+  'video/webm',
+  'video/quicktime',
+];
+
+export type AttachmentProps = {
+  /**
+   * The attachment to render
+   * See [Attachment structure](https://getstream.io/chat/docs/#message_format)
+   **/
+  attachments: ExtendedAttachment[];
+  /**
+   * Handle for click on action on Attachment
+   */
+  actionHandler?: (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    name?: string,
+    value?: string,
+  ) => void;
+  /**
+   * Custom UI component for attachment actions
+   * Defaults to [AttachmentActions](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Attachment/AttachmentActions.tsx)
+   */
+  AttachmentActions?: React.ComponentType<AttachmentActionsProps>;
+  /**
+   * Custom UI component for audio type attachment
+   * Defaults to [Audio](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Attachment/Audio.tsx)
+   */
+  Audio?: React.ComponentType<AudioProps>;
+  /**
+   * Custom UI component for card type attachment
+   * Defaults to [Card](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Attachment/Card.tsx)
+   */
+  Card?: React.ComponentType<CardProps>;
+  /**
+   * Custom UI component for file type attachment
+   * Defaults to [File](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Attachment/FileAttachment.tsx)
+   */
+  File?: React.ComponentType<FileAttachmentProps>;
+  /**
+   * Custom UI component for gallery type attachment
+   * Defaults to [AttachmentActions](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Gallery/Gallery.tsx)
+   */
+  Gallery?: React.ComponentType<GalleryProps>;
+  /**
+   * Custom UI component for image type attachment
+   * Defaults to [Image](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Gallery/Image.js)
+   */
+  Image?: React.ComponentType<ImageProps>;
+
+  /**
+   * Custom UI component for media type attachment
+   * Defaults to ReactPlayer from 'react-player'
+   */
+  Media?: React.ComponentType<ReactPlayerProps>;
+};
+
+export type BaseAttachmentUIComponentProps = {
+  /** The attachment to render */
+  /**
+		The handler function to call when an action is selected on an attachment.
+		Examples include canceling a \/giphy command or shuffling the results.
+		*/
+  actionHandler?: (
```

Comment:
something get messed up with this comment? do we need `attachment` too based on the first comment?

Reply:
Whoops, deleted that extra /** The attachment to render */.